### ISSUE TITLE
azurerm_kubernetes_cluster & kubernetes_cluster_node_pool: upgrade_settings add node_drain_timeout

### ISSUE THREAD

**Issue by norm-dole-dbtlabs** [hashicorp/terraform-provider-azurerm#24137]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Description

azurerm_kubernetes_cluster & kubernetes_cluster_node_pool: upgrade_settings currently has the attribute 'surge' but recently Microsoft added the ability to set 'node_drain_timeout' (default 30(minutes) with a max of 1440(1 day))

This allows AKS to respect pods with 'terminationGracePeriodSeconds' that exceed 30 minutes without failing the upgrade. This functionality was recently added and long overdue to allow automation to proceed through the upgrades without failing if there was a long running pod ex.) user ide pod or database run pod that exceeded 30 minutes.

Example)
az aks nodepool update -n node_pool_name -g resource_group --cluster-name cluster_name --drain-timeout 1440

https://learn.microsoft.com/en-us/rest/api/aks/agent-pools/upgrade-node-image-version?view=rest-aks-2023-08-01&tabs=Go

AgentPoolUpgradeSettings
Settings for upgrading an agentpool

Name	Type	Description
drainTimeoutInMinutes	
integer

The drain timeout for a node
The amount of time (in minutes) to wait on eviction of pods and graceful termination per node. This eviction wait time honors waiting on pod disruption budgets. If this time is exceeded, the upgrade fails. If not specified, the default is 30 minutes.

maxSurge	
string

The maximum number or percentage of nodes that are surged during upgrade.
This can either be set to an integer (e.g. '5') or a percentage (e.g. '50%'). If a percentage is specified, it is the percentage of the total agent pool size at the time of the upgrade. For percentages, fractional nodes are rounded up. If not specified, the default is 1. For more information, including best practices, see: https://docs.microsoft.com/azure/aks/upgrade-

### New or Affected Resource(s)/Data Source(s)

azurerm_kubernetes_cluster/kubernetes_cluster_node_pool

### Potential Terraform Configuration

```hcl
resource "azurerm_kubernetes_cluster" "aks" {
  name                              = "aks-${var.namespace}-${var.environment}"
  location                          = var.region
  resource_group_name               = var.resource_group_name
  default_node_pool {
    name                        = var.default_node_pool_name
    node_count                  = var.k8s_node_count
    vm_size                     = var.k8s_node_size
    temporary_name_for_rotation = "tempvmsize"
    vnet_subnet_id              = var.private_subnet_id

    upgrade_settings {
      max_surge = 3
      drain_timeout = 1440   ##### new section #############
    }
```

**Comment by cailyoung** [comment]
Is this a dupe of #13066?

**Comment by norm-dole-dbtlabs** [comment]
I think that's a good call out; I may be biased but this is a better/more complete implementation of #13066 but they are both seeking the same outcome of adding 'drain_timeout' 

# this is a worse implementation because 'max_surge'(existing) and 'drain_timeout' both belong under 'upgrade settings'
timeouts {
      drain = 40
  }

**Comment by rcskosir** [comment]
@cailyoung Thanks for spotting the duplication. 
@norm-dole-dbtlabs We like to try to keep discussions consolidated, so we’re going to close this new issue in favor of #13066. I encourage you to add your vote to the older issue as well as your suggestion of `upgrade_settings {
      max_surge = 3
      drain_timeout = 1440   ##### new section #############
    }` in a comment. Thanks again!
